pkg/common-controller/snapshot_controller.go

```diff
@@ -819,8 +819,17 @@ func (ctrl *csiSnapshotCommonController) ensurePVCFinalizer(snapshot *crdv1.Volu
 		return newControllerUpdateError(pvc.Name, "cannot add finalizer on claim because it is being deleted")
 	}
 
+	// Check to see if the backing PV is created by a CSI driver. If not, then remove the Finalizer,
+	// as the snapshot can't be created from this PVC
+	pv, err := ctrl.getVolumeFromVolumeSnapshot(snapshot)
```

Comment:
This could fail for other reasons.  This method does not really check if PV is backed by a CSI driver.